i-dot-ai/redbox-copilot
# Add Axe/Playwright tests

## Context

This is setting up Playwright for accessibility and end-to-end testing, using the pytest-playwright package. The accessibility tests use Axe.

The primary focus was accessibility testing, but this also serves as an MVP for additional end-to-end testing in future.


## Setup

Follow the instructions in the README at `django_app/tests_playwright/`.

If any pages aren't working on this branch (.e.g. /documents) you may need to run `poetry run python manage.py migrate`. 


## How to test

1. First check that all the tests pass

2. Then create some accessibility issues on a page. Here's a few if you're struggling for ideas:
```
<h2>Main page heading, instead of a h1</h2>
<a href=""></a>
<img src="broken-and-no-alt.jpg"/>
```
And check the errors are reported
